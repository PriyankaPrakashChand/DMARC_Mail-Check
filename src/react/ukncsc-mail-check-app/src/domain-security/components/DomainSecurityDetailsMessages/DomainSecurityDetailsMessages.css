.DomainSecuritySummaryMessages--intro {
  padding: 15px 0px 10px;
  font-weight: 600;
}
