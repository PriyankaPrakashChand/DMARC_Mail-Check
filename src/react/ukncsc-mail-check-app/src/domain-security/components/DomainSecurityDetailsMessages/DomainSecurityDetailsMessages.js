import React from 'react';
import pluralise from 'pluralize';
import PropTypes from 'prop-types';
import { Divider } from 'semantic-ui-react';
import { DomainSecurityPill } from 'domain-security/components';
import { MailCheckMessage } from 'common/components';
import './DomainSecurityDetailsMessages.css';

const DomainSecurityDetailsMessages = ({
  type,
  markdown,
  failures,
  warnings,
  inconclusives,
}) => (
  <React.Fragment>
    {!!failures.length && (
      <React.Fragment>
        <DomainSecurityPill error>{failures.length} Urgent</DomainSecurityPill>
        <p className="DomainSecuritySummaryMessages--intro">
          Your {type} configuration has{' '}
          {pluralise('element', failures.length, true)} that{' '}
          {pluralise.singular('require', failures.length)} urgent attention.
        </p>
        {failures.map((description, i) => (
          <MailCheckMessage error markdown={markdown} key={i}>
            {description}
          </MailCheckMessage>
        ))}
        <Divider hidden />
      </React.Fragment>
    )}
    {!!warnings.length && (
      <React.Fragment>
        <DomainSecurityPill warning>
          {warnings.length} Advisory
        </DomainSecurityPill>
        <p className="DomainSecuritySummaryMessages--intro">
          Your {type} configuration has{' '}
          {pluralise('element', warnings.length, true)} that can be improved.
        </p>
        {warnings.map((description, i) => (
          <MailCheckMessage warning markdown={markdown} key={i}>
            {description}
          </MailCheckMessage>
        ))}
        <Divider hidden />
      </React.Fragment>
    )}
    {!!inconclusives.length && (
      <React.Fragment>
        <DomainSecurityPill information>
          {inconclusives.length} Inconclusive
        </DomainSecurityPill>
        <p className="DomainSecuritySummaryMessages--intro">
          Your {type} configuration has {inconclusives.length} inconclusive{' '}
          {pluralise('element', inconclusives.length)}.
        </p>
        {inconclusives.map((description, i) => (
          <MailCheckMessage markdown={markdown} key={i}>
            {description}
          </MailCheckMessage>
        ))}
      </React.Fragment>
    )}
    {!failures.length &&
      !warnings.length &&
      !inconclusives.length && (
        <React.Fragment>
          <DomainSecurityPill>1 Positive</DomainSecurityPill>
          <Divider hidden />
          <MailCheckMessage success>{type} well configured.</MailCheckMessage>
        </React.Fragment>
      )}
  </React.Fragment>
);

DomainSecurityDetailsMessages.defaultProps = {
  markdown: false,
  failures: [],
  warnings: [],
  inconclusives: [],
};

DomainSecurityDetailsMessages.propTypes = {
  type: PropTypes.string.isRequired,
  markdown: PropTypes.bool,
  failures: PropTypes.arrayOf(PropTypes.string),
  warnings: PropTypes.arrayOf(PropTypes.string),
  inconclusives: PropTypes.arrayOf(PropTypes.string),
};

export default DomainSecurityDetailsMessages;
