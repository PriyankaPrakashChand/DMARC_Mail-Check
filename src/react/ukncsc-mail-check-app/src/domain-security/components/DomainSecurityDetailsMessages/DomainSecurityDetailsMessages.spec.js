import React from 'react';
import { render } from 'react-testing-library';
import DomainSecurityDetailsMessages from './DomainSecurityDetailsMessages';

describe('DomainSecurityDetailsMessages', () => {
  let container;

  describe('when provided with no failures, warnings or inconclusives', () => {
    beforeEach(() => {
      ({ container } = render(<DomainSecurityDetailsMessages type="DMARC" />));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should display a success message', () =>
      expect(
        container.getElementsByClassName('ui success message')
      ).toHaveLength(1));
  });

  describe('when provided with failures', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityDetailsMessages type="DMARC" failures={['oh noes']} />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should display a failure message', () =>
      expect(container.getElementsByClassName('ui error message')).toHaveLength(
        1
      ));

    test('it should not display any warnings messages', () =>
      expect(
        container.getElementsByClassName('ui warning message')
      ).toHaveLength(0));
  });

  describe('when provided with warnings', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityDetailsMessages type="DMARC" warnings={['oh woes']} />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should display a warning message', () =>
      expect(
        container.getElementsByClassName('ui warning message')
      ).toHaveLength(1));

    test('it should not display any failure messages', () =>
      expect(container.getElementsByClassName('ui error message')).toHaveLength(
        0
      ));
  });

  describe('when provided with inconclusives', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityDetailsMessages
          type="DMARC"
          inconclusives={['who knows']}
        />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should not display any warning messages', () =>
      expect(
        container.getElementsByClassName('ui warning message')
      ).toHaveLength(0));

    test('it should not display any failure messages', () =>
      expect(container.getElementsByClassName('ui error message')).toHaveLength(
        0
      ));
  });

  describe('when provided with failures, warnings and inconclusives', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityDetailsMessages
          type="DMARC"
          failures={['oh noes']}
          warnings={['oh woes']}
          inconclusives={['who knows']}
        />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should display a failure message', () =>
      expect(container.getElementsByClassName('ui error message')).toHaveLength(
        1
      ));

    test('it should display a warning message', () =>
      expect(container.getElementsByClassName('ui error message')).toHaveLength(
        1
      ));
  });

  describe('when provided with a markdown flag', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecurityDetailsMessages
          markdown
          type="DMARC"
          failures={['[link](https://ncsc.gov.uk)']}
        />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('it should render the text as markdown', () =>
      expect(container.getElementsByTagName('a')).toHaveLength(1));
  });
});
