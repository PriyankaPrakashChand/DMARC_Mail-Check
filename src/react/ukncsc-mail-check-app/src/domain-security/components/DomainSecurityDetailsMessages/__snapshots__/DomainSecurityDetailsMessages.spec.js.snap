// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainSecurityDetailsMessages when provided with a markdown flag it should match the snapshot 1`] = `
<div>
  <div
    class="ui red large circular label"
  >
    <span
      class="DomainSecurityPill"
    >
      1
      Urgent
    </span>
  </div>
  <p
    class="DomainSecuritySummaryMessages--intro"
  >
    Your 
    DMARC
    configuration has
    1 element
    that
    require
    urgent attention.
  </p>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui error message MailCheckMessage--container"
  >
    <div>
      <p>
        <a
          href="https://ncsc.gov.uk"
        >
          link
        </a>
      </p>
    </div>
  </div>
  <div
    class="ui hidden divider"
  />
</div>
`;

exports[`DomainSecurityDetailsMessages when provided with failures it should match the snapshot 1`] = `
<div>
  <div
    class="ui red large circular label"
  >
    <span
      class="DomainSecurityPill"
    >
      1
      Urgent
    </span>
  </div>
  <p
    class="DomainSecuritySummaryMessages--intro"
  >
    Your 
    DMARC
    configuration has
    1 element
    that
    require
    urgent attention.
  </p>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui error message MailCheckMessage--container"
  >
    oh noes
  </div>
  <div
    class="ui hidden divider"
  />
</div>
`;

exports[`DomainSecurityDetailsMessages when provided with failures, warnings and inconclusives it should match the snapshot 1`] = `
<div>
  <div
    class="ui red large circular label"
  >
    <span
      class="DomainSecurityPill"
    >
      1
      Urgent
    </span>
  </div>
  <p
    class="DomainSecuritySummaryMessages--intro"
  >
    Your 
    DMARC
    configuration has
    1 element
    that
    require
    urgent attention.
  </p>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui error message MailCheckMessage--container"
  >
    oh noes
  </div>
  <div
    class="ui hidden divider"
  />
  <div
    class="ui yellow large circular label"
  >
    <span
      class="DomainSecurityPill"
    >
      1
      Advisory
    </span>
  </div>
  <p
    class="DomainSecuritySummaryMessages--intro"
  >
    Your 
    DMARC
    configuration has
    1 element
    that can be improved.
  </p>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui warning message MailCheckMessage--container"
  >
    oh woes
  </div>
  <div
    class="ui hidden divider"
  />
  <div
    class="ui grey large circular label"
  >
    <span
      class="DomainSecurityPill"
    >
      1
      Inconclusive
    </span>
  </div>
  <p
    class="DomainSecuritySummaryMessages--intro"
  >
    Your 
    DMARC
    configuration has 
    1
    inconclusive
    element
    .
  </p>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui message MailCheckMessage--container"
  >
    who knows
  </div>
</div>
`;

exports[`DomainSecurityDetailsMessages when provided with inconclusives it should match the snapshot 1`] = `
<div>
  <div
    class="ui grey large circular label"
  >
    <span
      class="DomainSecurityPill"
    >
      1
      Inconclusive
    </span>
  </div>
  <p
    class="DomainSecuritySummaryMessages--intro"
  >
    Your 
    DMARC
    configuration has 
    1
    inconclusive
    element
    .
  </p>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui message MailCheckMessage--container"
  >
    who knows
  </div>
</div>
`;

exports[`DomainSecurityDetailsMessages when provided with no failures, warnings or inconclusives it should match the snapshot 1`] = `
<div>
  <div
    class="ui green large circular label"
  >
    <span
      class="DomainSecurityPill"
    >
      1 Positive
    </span>
  </div>
  <div
    class="ui hidden divider"
  />
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui success message MailCheckMessage--container"
  >
    DMARC
    well configured.
  </div>
</div>
`;

exports[`DomainSecurityDetailsMessages when provided with warnings it should match the snapshot 1`] = `
<div>
  <div
    class="ui yellow large circular label"
  >
    <span
      class="DomainSecurityPill"
    >
      1
      Advisory
    </span>
  </div>
  <p
    class="DomainSecuritySummaryMessages--intro"
  >
    Your 
    DMARC
    configuration has
    1 element
    that can be improved.
  </p>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui warning message MailCheckMessage--container"
  >
    oh woes
  </div>
  <div
    class="ui hidden divider"
  />
</div>
`;
